### PR TITLE
Menu Scenes

### PR DESCRIPTION
**Pull request by emilie-martin**
image

* implemented a main menu (with controls overlay/canvas)
> haven't done controller mapping but if someone wants to provide me withe the right button mapping I could add it in easily
* implemented a game over menu (there's a placeholder bcs idk what other button to put)
* implemented basic scripting for buttons clicks/presses
* implemented a canvas/overlay to load saves (didn't implement functionality as i wasn't too sure how so both just start the dev scene as is)

> **note:** we might want to have a "loading" scene implemented later on bcs there's a noticeable delay (on my end at least) when you hit play from the main menu

### CODE REVIEW

File: Hush/Assets/Scripts/Game/Menus.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        TimeManager.Instance.Pause();
+        EventSystem.current.SetSelectedGameObject(pauseFirstButton);
+
+        pauseMenuUI.SetActive(true);
+    }
+
+    public void Resume()
+    {
+        Debug.Log("Resume");
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Game/Menus.cs:33]: Clear out debug logs here

**Review by daniela-venuta — CHANGES_REQUESTED** [review]
Null reference exception when clicking on Play. Otherwise looks amazing :)

**Review by arejayelle — CHANGES_REQUESTED** [review]
please, set Canvases to scale w/ screensize
| Gameover | Menu |
|--|--|
|image | image|

## Main Menu
- [ ] Add `SceneManager`  prefab to the scene to avoid null reference
- [ ] Make "Play" button more prominent than "Controls" button. 
### Minor
- [ ] Center the Buttons object
- [ ] Set Child Alignment to `Middle Center`
- [ ] Make the Recttransform itself wider
- [ ] Underline isn't available for Subtitle

## Gameover
- [ ] Add `SceneManager`  prefab to the scene to avoid null reference
- [ ] Gameover menu should be able to go back to Main Menu & quit
- [ ] Please click the magic button to fix the input issues (on `EventSystem`)
  *image

File: Hush/Assets/Scripts/Game/Menus.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public void Credits() {
+        // POP-UP FOR CREDITS
+        Debug.Log("CREDITS");
+    }
+    
+    public void MainMenu() {
+        Game.SceneManager.Instance.LoadMainMenu();
+        Debug.Log("BACK TO MAIN MENU");
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Game/Menus.cs:45]: ### minor
Can be removed since the function has been implemented

File: Hush/Assets/Scripts/Game/Menus.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public void Quit()
+    {
+        Debug.Log("QUIT");
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Game/Menus.cs:50]: ### minor
Can be removed since the function has been implemented

File: Hush/Assets/Scripts/Game/SceneManager.cs [hunk 4]
```diff
@@ -26,15 +29,25 @@ public void LoadNextScene()
...
-        public void LoadDemoScene()
+        public void LoadDevScene()
         {
             UnityEngine.SceneManagement.SceneManager.LoadScene(1);
         }
+
+        public void LoadDemoScene()
+        {
+            UnityEngine.SceneManagement.SceneManager.LoadScene(2);
+        }
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Game/SceneManager.cs:51]: ### Minor

Arguably this code can be removed since we shouldn't need to access the demo scene anymore

File: Hush/Assets/Scenes/Menus/MainMenu.unity [hunk 5]
```diff
@@ -0,0 +1,4942 @@
+%YAML 1.1
```
Comment (arejayelle) [inline Hush/Assets/Scenes/Menus/MainMenu.unity:1]: For consistency, keep the buttons font for buttons. (you can use the buttons font for text here if you want, but not the other way around)


### minor
1. How many save files per "install" will we support? Just one? Mutliple? The current interface for it feels a bit off. Either we make save files A, B C or we just have a single file? eiother way, here's a mockup
image
 
2. add a black panel w/ opacity (like alpha = 75% ish ) dim down the background and make this menu stand out more. 
image

File: Hush/Assets/Scenes/Menus/GameOver.unity [hunk 6]
```diff
@@ -0,0 +1,2127 @@
+%YAML 1.1
```
Comment (arejayelle) [inline Hush/Assets/Scenes/Menus/GameOver.unity:1]: Make the buttons bigger and change the text (though you can also do this is another PR since Siobhan needs these changes)

image

**Review by arejayelle — CHANGES_REQUESTED** [review]
Some more changes, but if you wanna do them in another PR that's fine too. Also is the You died screen supposed to be hooked up in this or just implemneted and left for another PR? It's fine if it is, i'm just clarifying

File: Hush/Assets/Scripts/Game/Menus.cs (hunk 3 above)
Comment (emilie-martin) [reply Hush/Assets/Scripts/Game/Menus.cs:49]: i might just keep this one because it was there when we're running the project out of build, so we can actually see when it's being called properly

File: Hush/Assets/Scenes/Menus/MainMenu.unity (hunk 5 above)
Comment (emilie-martin) [reply Hush/Assets/Scenes/Menus/MainMenu.unity:1]: i'm not super clear on that either. from what i understood it was just a single one (wherein a `Start` vs `Continue` might make more sense, if that's what you mean) but I don't know if we're actually considering multiple save files. if we want multiple save files we can make task for it and implement it in another PR

**Review by daniela-venuta — APPROVED** [review]

**Review by arejayelle — APPROVED** [review]
LGTM
